### PR TITLE
SDI-508 Handle activities.prisoner.deallocated event

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt [hunk 1]
```diff
@@ -110,6 +111,39 @@ class ActivitiesService(
...
+              // endComment = allocation.?, // TODO could put something useful in here
+            )
+          )
+        } catch (e: Exception) {
+          telemetryClient.trackEvent("activity-deallocate-failed", telemetryMap)
+          throw e
+        }
+
+        val mapWithNomisId = telemetryMap
+          .plus(Pair("offenderProgramReferenceId", nomisResponse.offenderProgramReferenceId.toString()))
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt:140]: Why re-assign to a new value given the map is mutable?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt [hunk 2]
```diff
@@ -246,6 +246,84 @@ internal class ActivitiesServiceTest {
...
+
+      activitiesService.deallocate(
+        AllocationDomainEvent(
+          eventType = "dummy",
+          scheduleId = activityScheduleId,
+          allocationId = allocationId,
+          version = "1.0",
+          description = "description",
+          occurredAt = LocalDateTime.now(),
+        )
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt:275]: I guess it doesn't matter as the events haven't been defined by the producer yet, but this doesn't fit the domain event schema - scheduleId and allocationId should be in an `additionalInformation` section and there should be a detail URL that points back to the Activities API.

I'd be slightly worried that the Activities team just copy this structure when they get around to producing events.

**Review by mikehalmamoj — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt (hunk 1 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt]: good point, have changed to just put to the existing map

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt (hunk 2 above)
Comment (steverendell) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesServiceTest.kt]: Done

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**